calculators/referendum_expenditures_by_origin.rb

```diff
@@ -0,0 +1,101 @@
+      committees.where('"Filer_ID" IS NOT NULL').index_by { |c| c.Filer_ID }
+  end
+
+  def fetch
+    totals = ActiveRecord::Base.connection.execute(<<-SQL)
+      SELECT "Measure_Number", "Sup_Opp_Cd", sum("Amount") AS total
+      FROM "efile_COAK_2016_E-Expenditure",
+        oakland_name_to_number
+      WHERE "Bal_Name" = "Measure_Name"
+      GROUP BY "Measure_Number", "Sup_Opp_Cd";
+    SQL
+    expenditures = ActiveRecord::Base.connection.execute(<<-SQL)
+      SELECT d."Measure_Number", d."Sup_Opp_Cd", A.locale, A.total
+      FROM
+        (SELECT distinct "Filer_ID", "Measure_Number", "Sup_Opp_Cd"
+          FROM "efile_COAK_2016_E-Expenditure",
+          oakland_name_to_number
+          WHERE "Bal_Name" IS NOT NULL
+          AND "Bal_Name" = "Measure_Name") d,
+        (SELECT "Filer_ID",
+        CASE
+          WHEN lower("Tran_City") = 'oakland' THEN 'Within Oakland'
+          WHEN upper("Tran_State") = 'CA' THEN 'Within California'
+          ELSE 'Out of State'
+        END as locale,
+        sum("Tran_Amt1") as total
+        FROM
+          (SELECT "Filer_ID", "Tran_City", "Tran_State", "Tran_Amt1"
+          FROM "efile_COAK_2016_A-Contributions"
+          UNION
+          SELECT CAST("Filer_ID" as VARCHAR(7)),
+          "Enty_City" as "Tran_City", "Enty_ST" as "Tran_State",
+          "Amount" as "Tran_Amt1"
+          FROM "efile_COAK_2016_497") x
+        WHERE "Filer_ID" IN (SELECT "Filer_ID"
+              FROM "efile_COAK_2016_E-Expenditure"
+              WHERE "Bal_Name" IS NOT NULL)
+        GROUP BY "Filer_ID", locale) A
+      WHERE d."Filer_ID" = A."Filer_ID"
+      ORDER BY d."Measure_Number", d."Sup_Opp_Cd";
+    SQL
+
+    support_total = {}
+    oppose_total = {}
+
+    totals.each do |row|
+      if row['Sup_Opp_Cd'] == 'S'
+        support_total[row['Measure_Number']] = row['total']
+      elsif row['Sup_Opp_Cd'] == 'O'
+        oppose_total[row['Measure_Number']] = row['total']
+      end
+    end
+
+    support = {}
+    oppose = {}
+
+    expenditures.each do |row|
+      if row['Sup_Opp_Cd'] == 'S'
+        support[row['Measure_Number']] ||= {}
+        support[row['Measure_Number']][row['locale']] = row['total']
+      elsif row['Sup_Opp_Cd'] == 'O'
+        oppose[row['Measure_Number']] ||= {}
+        oppose[row['Measure_Number']][row['locale']] = row['total']
+      end
+    end
+
+    [
+      [support_total, support, :supporting_locales],
+      [oppose_total, oppose, :opposing_locales],
+    ].each do |totals, locales, calculation_name|
+      totals.keys.each do |measure|
+        ballot_measure = ballot_measure_from_number(measure)
+        result = locales[measure].keys.map do |locale|
+          amount = locales[measure][locale]
+          totals[measure] -= amount
+          {
+            locale: locale,
+            amount: amount,
+          }
+        end
+        if totals[measure] > 0
+          result <<
+          {
+            locale: 'unknown',
```

Comment:
Can you capitalize all the strings in this file, so the frontend can display them without needing to do anything?